internal/applicationsnapshot/input.go

```diff
@@ -97,6 +112,8 @@ func DetermineInputSpec(ctx context.Context, input Input) (*app.SnapshotSpec, er
 		if err != nil {
 			return nil, err
 		}
+		log.Debugf("Calling expandImageIndex")
+		expandImageIndex(ctx, &file)
```

Comment:
Could expandImageIndex be a function on the `snapshot` struct? Either way calling it once at the end of `DetermineInputSpec` should suffice, having it in each branch here makes it difficult to maintain and could mistakenly lead to doing more work than necessary by mistake (expanding the image index multiple times for the same image index)